File: src/test/java/io/bloco/faker/helpers/StringHelperTest.java
package io.bloco.faker.helpers;

import org.junit.Before;
import org.junit.Test;

import java.util.regex.Matcher;

import static org.hamcrest.core.Is.is;
import static org.hamcrest.core.IsEqual.equalTo;
import static org.junit.Assert.assertThat;

public class StringHelperTest {

    private StringHelper stringHelper;

    @Before
    public void setUp() {
        stringHelper = new StringHelper();
    }

    @Test
    public void testReplaceMethod() {
        assertThat(stringHelper.replaceMethod("aaa", ".", new StringHelper.StringReplacer() {
            @Override
            public String replaceWith(Matcher matcher) {
                return "b";
            }
        }), is(equalTo("bbb")));

        assertThat(stringHelper.replaceMethod("abc", "a(.)", new StringHelper.StringReplacer() {
            @Override
            public String replaceWith(Matcher matcher) {
                return matcher.group(1).toUpperCase();
            }
        }), is(equalTo("Bc")));
    }

    @Test
    public void testSnakeToCamel() {
        assertThat(stringHelper.snakeToCamel("hello"), is(equalTo("hello")));
        assertThat(stringHelper.snakeToCamel("first_name"), is(equalTo("firstName")));
        assertThat(stringHelper.snakeToCamel("once_upon_a_time"), is(equalTo("onceUponATime")));
    }

    @Test
    public void testCamelToSnake() {
        assertThat(stringHelper.camelToSnake("HelloWorld"), is(equalTo("hello_world")));
        assertThat(stringHelper.camelToSnake("SlackEmoji"), is(equalTo("slack_emoji")));
        assertThat(stringHelper.camelToSnake("RGB"), is(equalTo("r_g_b")));
        assertThat(stringHelper.camelToSnake("Rgb"), is(equalTo("rgb")));
    }

    @Test
    public void testNormalize() {
        assertThat(stringHelper.normalize("Sérgio Santos"), is(equalTo("sergiosantos")));
        assertThat(stringHelper.normalize("áàãâäå"), is(equalTo("aaaaaa")));
        assertThat(stringHelper.normalize("-.@,;!?"), is(equalTo("")));
    }
}

File: src/test/kotlin/io/bloco/faker/helpers/StringHelperTest.kt
package io.bloco.faker.helpers

import junit.framework.TestCase.assertTrue
import org.junit.Test

class StringHelperTest {

    @Test
    fun testSnakeToCamel() {
        assertTrue("hello".snakeToCamel() == "hello")
        assertTrue("first_name".snakeToCamel() == "firstName")
        assertTrue("once_upon_a_time".snakeToCamel() == "onceUponATime")
    }

    @Test
    fun testCamelToSnake() {
        assertTrue("HelloWorld".camelToSnake() == "hello_world")
        assertTrue("SlackEmoji".camelToSnake() == "slack_emoji")
        assertTrue("RGB".camelToSnake() == "r_g_b")
        assertTrue("Rgb".camelToSnake() == "rgb")
    }

    @Test
    fun testNormalize() {
        assertTrue("Sérgio Santos".normalize() == "sergiosantos")
        assertTrue("áàãâäå".normalize() == "aaaaaa")
        assertTrue("-.@,;!?".normalize() == "")
    }
}
